fixed-requirements.txt

```diff
@@ -27,11 +27,12 @@ lockfile==0.12.2
 MarkupSafe<2.1.0,>=0.23
 mongoengine==0.23.0
 # networkx v2.6 does not support Python3.6.  Update networkx to match orquesta
-networkx>=2.5.1,<2.6
+networkx>=2.5.1,<2.6; python_version < '3.7'
+networkx>=2.6<3; python_version >= '3.7'
```

Comment:
The python version condition in requirements comes from `orquesta` here (https://github.com/StackStorm/orquesta/pull/255/files) and will be a difficulty of this PR which fails due to duplicate `networkx` records: https://app.circleci.com/pipelines/github/StackStorm/st2/4153/workflows/0b3b950e-38de-4056-be6c-d1e4a05ff2d8/jobs/15935

```
[bdist: st2api] [20:19:54]      /usr/bin/python3 ../scripts/fixate-requirements.py -s in-requirements.txt -f ../fixed-requirements.txt
[bdist: st2api] [20:19:55]      Traceback (most recent call last):
[bdist: st2api] [20:19:55]        File "../scripts/fixate-requirements.py", line 284, in <module>
[bdist: st2api] [20:19:55]          skip=skip,
[bdist: st2api] [20:19:55]        File "../scripts/fixate-requirements.py", line 195, in write_requirements
[bdist: st2api] [20:19:55]          'Duplicate definition for dependency "%s"' % (project_name)
[bdist: st2api] [20:19:55]      ValueError: Duplicate definition for dependency "networkx"
[bdist: st2api] [20:19:55]      Makefile:74: recipe for target '.stamp-requirements' failed
[bdist: st2api] [20:19:55]      make: *** [.stamp-requirements] Error 1
```

Looks like we don't support it
https://github.com/StackStorm/st2/blob/8f6cb468ff50a827ce50c9be6ba78b4ac90da263/scripts/fixate-requirements.py#L193-L198 and relying on a dict for dependency names additionally checking for duplicates.

@cognifloyd @nzlosh Thinking how to make it work 🤔 Do you know any hacks with the @git-like links or non-standard formats in requirements we can possibly use?